File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/ReciprocalBackFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.SettledBet;
import cz.fb.manaus.core.model.Side;
import org.springframework.stereotype.Component;

import java.util.OptionalDouble;

@Component
public class ReciprocalBackFunction implements ProgressFunction {

    @Override
    public OptionalDouble apply(SettledBet bet) {
        return bet.getBetAction().getMarketPrices().getReciprocal(Side.BACK);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/ReciprocalBackFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.RealizedBet
import cz.fb.manaus.core.model.Side
import cz.fb.manaus.reactor.price.getReciprocal
import org.springframework.stereotype.Component

@Component
object ReciprocalBackFunction : ProgressFunction {

    override fun invoke(bet: RealizedBet): Double? {
        return getReciprocal(bet.betAction.runnerPrices, Side.BACK)
    }

}
